cli/deno_dir.rs

```diff
@@ -1240,6 +1301,50 @@ mod tests {
     });
   }
 
+  #[test]
+  fn test_get_source_code_multiple_downloads_of_same_file() {
+    let (_temp_dir, deno_dir) = test_setup();
+    // http_util::fetch_sync_string requires tokio
+    tokio_util::init(|| {
+      let module_name = "http://localhost:4545/tests/subdir/mismatch_ext.ts";
+      let filename = deno_fs::normalize_path(
+        deno_dir
+          .deps_http
+          .join("localhost_PORT4545/tests/subdir/mismatch_ext.ts")
+          .as_ref(),
+      );
+      let headers_file_name = source_code_headers_filename(&filename);
+
+      // first download
+      let result =
+        get_source_code(&deno_dir, module_name, &filename, false, false);
+      assert!(result.is_ok());
+
+      let result = fs::File::open(&headers_file_name);
+      assert!(result.is_ok());
+      let headers_file = result.unwrap();
+      // save modified timestamp for headers file
+      let headers_file_metadata = headers_file.metadata().unwrap();
+      let headers_file_modified = headers_file_metadata.modified().unwrap();
+
+      // download file again, it should use already fetched file even though `use_cache` is set to
+      // false, this can be verified using source header file creation timestamp (should be
+      // the same as after first download)
+      let result =
+        get_source_code(&deno_dir, module_name, &filename, false, false);
+      assert!(result.is_ok());
+
+      let result = fs::File::open(&headers_file_name);
+      assert!(result.is_ok());
+      let headers_file_2 = result.unwrap();
+      // save modified timestamp for headers file
+      let headers_file_metadata_2 = headers_file_2.metadata().unwrap();
+      let headers_file_modified_2 = headers_file_metadata_2.modified().unwrap();
+
+      assert_eq!(headers_file_modified, headers_file_modified_2);
```

Comment:
Very nice! I've been trying to come up with a test for this failure for a long time.

Thank you for solving this bug!

Reply:
> Very nice! I've been trying to come up with a test for this failure for a long time. I have verified this test does indeed fail when applied to master.
> 
> Thank you for solving this bug!

Great, thanks! :tada: